### PR TITLE
Creating pull request for partial solution to aiport challenge

### PR DESCRIPTION
**Pull request by fg24davies**
updated with README.md with how I found the challenge

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,56 +1,37 @@
-Airport Challenge
-=================
+# Airport Challenge - Weekend 1
+I found this really challenging and have submitted a partial solution. The parts that felt similar to Boris Bike Challenge were doable but I did struggle beyond that.
```
Comment (eatruman) [inline README.md:2]: I felt the same way!

File: spec/airport_spec.rb [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+  it 'can take off plane' do
+    expect(subject).to respond_to(:takeoff)
+  end
+  
+  context 'weather is good' do 
+    before do
+      allow(subject).to receive(:stormy?) { false }
+    end
+    it 'raises an error when full' do 
+      subject.land(Plane.new)
```
Comment (eatruman) [inline spec/airport_spec.rb:21]: If you created a plane underneath your describe Airport do, you could then use that  in all of these tests rather than using (Plane.new) each time.

File: spec/airport_spec.rb [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+      expect { Airport::DEFAULT_CAPACITY.times do
+        subject.land(Plane.new) 
+      end 
+      }.to raise_error 'Airport is full: plane cannot land'
+    end 
+
+    it 'raises an error when empty' do
+      expect { subject.takeoff(Plane.new) }.to raise_error 'Airport is empty: no planes to takeoff'
```
Comment (eatruman) [inline spec/airport_spec.rb:29]: Do you need a line before your expect to make sure your planes_at_airport array is empty before testing this? Otherwise you're just expecting calling takeoff with a new plane to return that the airport is empty when it might not be?

File: spec/airport_spec.rb [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+  end
+
+  it 'has a default capacity' do
+    expect(subject.capacity).to eq Airport::DEFAULT_CAPACITY
+  end
+
+  
+
+  context 'weather is stormy' do
+    before do
```
Comment (eatruman) [inline spec/airport_spec.rb:40]: This was a good idea rather than setting Stormy? to true in each test, it makes your code cleaner and easy to read/understand.

File: lib/plane.rb [hunk 5]
```diff
@@ -0,0 +1,19 @@
+class Plane
+  attr_accessor :in_air
+  def initialize
+    @in_air = true 
+  end
+
+  def in_air?
+    @in_air
+  end
```
Comment (eatruman) [inline lib/plane.rb:9]: This was a good way to solve the location of the plane problem.

File: lib/weather.rb [hunk 6]
```diff
@@ -0,0 +1,5 @@
+module WeatherConditions 
+  def stormy?
+    rand(10) > 8
+  end
```
Comment (eatruman) [inline lib/weather.rb:4]: This is really clear and concise!

**Review by eatruman — COMMENTED** [review]
I think overall you did a great job! There were parts of the challenge missing but you completed it in order so you covered all of the user stories and started on the edge cases. I think it was clear what you were testing and your tests were easy to understand. Your code is also clear and easy to read. Going forward, you could improve by creating your own user stories for the edge cases to practice writing tests that are different from ones we have already done (which is what I will be working on too!) Good job Flora!
